intel_extension_for_transformers/llm/runtime/graph/core/layers/mha_dense.cpp

```diff
@@ -961,31 +961,46 @@ class ScaleTrackMax<ISA_T, float, float> {
     int causal_offset;  // offset for causal mask; negative value for disabling causal mask
     float alibi_slope;  // m-factor in the alibi paper for current head: https://arxiv.org/abs/2108.12409
   };
+  static constexpr float seq15[16]{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15};
 
   JBLAS_CODE forward(const SType* src, const int src_step, const int M_offset, const int N_offset, const int M,
                      const int N, const Param& p) const {
+    return p.alibi_slope == 0 ? forward_<false>(src, src_step, M_offset, N_offset, M, N, p)
+                              : forward_<true>(src, src_step, M_offset, N_offset, M, N, p);
+  }
+
+  template <bool HAS_ALIBI>
+  JBLAS_CODE forward_(const SType* src, const int src_step, const int M_offset, const int N_offset, const int M,
+                      const int N, const Param& p) const {
     const auto dst = p.dst + M_offset * p.ld_dst + N_offset;
     const auto dst_max = p.dst_max + M_offset;
 #if CompileAVX512F()
 #if MHA_2ND_EXP
     const auto v_scale = _mm512_set1_ps(p.scale);
+    const auto v_seq15 = _mm512_loadu_ps(seq15);
```

Comment:
I wonder whether these will infect performance if ```forward<false>```, will compiler allocate registers for these?

Reply:
I don't think so. Everything except `alibi_curr` are not used in the innermost loop. In addition, local performance test && CI shows that there is no performance regression on models without AliBi (e.g. llama).